### PR TITLE
chore(utils): Update base repository to infer primary keys and support composite

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Add support for composite keys support in the base module repository class

### CODE REVIEW

**Comment by adrien2p** [comment]
@olivermrbl can we merge this ?

File: packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts [hunk 1]
```diff
@@ -250,42 +257,71 @@ export function mikroOrmBaseRepositoryFactory<
...
+
       if (missingEntities.length) {
         const entityName = (entity as EntityClass<T>).name ?? entity
+        const missingEntitiesKeys = data.map((data_) =>
+          primaryKeys.map((key) => data_[key]).join(", ")
+        )
         throw new MedusaError(
           MedusaError.Types.NOT_FOUND,
-          `${entityName} with ${[primaryKey]} "${missingEntities.join(
+          `${entityName} with ${primaryKeys.join(
```
Comment (olivermrbl) [inline packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts:318]: nit:
```suggestion
          `${entityName} with primary key(s) ${primaryKeys.join(
```

**Review by olivermrbl — APPROVED** [review]
LGTM, one suggestion. Take it or leave it :)

File: packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts (hunk 1 above)
Comment (adrien2p) [reply packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts:318]: are you sure you prefer `Something with primary key(s) id` over `Something with id` - `Something with primary key(s) id, code` over `Something with id, code`

Comment (olivermrbl) [reply packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts:318]: I am indifferent. Fine to keep as is :)

Comment (adrien2p) [reply packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts:318]: I will keep it as is for now then :)

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

**Pull request merged**